source/ranges.tex

```diff
@@ -7430,3 +7476,2389 @@
+
+    template<bool OtherConst>
+      requires @\libconcept{sized_sentinel_for}@<sentinel_t<@\exposid{Base}@>, iterator_t<@\exposid{maybe-const}@<OtherConst, V>>>
+    friend constexpr range_difference_t<@\exposid{maybe-const}@<OtherConst, V>>
+      operator-(const @\exposid{sentinel}@& y, const @\exposid{iterator}@<OtherConst>& x);
+  };
+}
+\end{codeblock}
+
+\begin{itemdecl}
+constexpr explicit @\exposid{sentinel}@(sentinel_t<@\exposid{Base}@> end);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Initializes \exposid{end_} with \tcode{end}.
+\end{itemdescr}
+
+\begin{itemdecl}
+constexpr @\exposid{sentinel}@(@\exposid{sentinel}@<!Const> i)
+  requires Const && @\libconcept{convertible_to}@<sentinel_t<V>, sentinel_t<@\exposid{Base}@>>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Initializes \exposid{end_} with \tcode{std::move(i.\exposid{end_})}.
+\end{itemdescr}
+
+\begin{itemdecl}
+template<bool OtherConst>
+  requires @\libconcept{sentinel_for}@<sentinel_t<@\exposid{Base}@>, iterator_t<@\exposid{maybe-const}@<OtherConst, V>>>
+friend constexpr bool operator==(const @\exposid{iterator}@<OtherConst>& x, const @\exposid{sentinel}@& y);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to: \tcode{return x.\exposid{current_}.back() == y.\exposid{end_};}
+\end{itemdescr}
+
+\begin{itemdecl}
+template<bool OtherConst>
+  requires @\libconcept{sized_sentinel_for}@<sentinel_t<@\exposid{Base}@>, iterator_t<@\exposid{maybe-const}@<OtherConst, V>>>
+friend constexpr range_difference_t<@\exposid{maybe-const}@<OtherConst, V>>
+  operator-(const @\exposid{iterator}@<OtherConst>& x, const @\exposid{sentinel}@& y);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to: \tcode{return x.\exposid{current_}.back() - y.\exposid{end_};}
+\end{itemdescr}
+
+\begin{itemdecl}
+template<bool OtherConst>
+  requires @\libconcept{sized_sentinel_for}@<sentinel_t<@\exposid{Base}@>, iterator_t<@\exposid{maybe-const}@<OtherConst, V>>>
+friend constexpr range_difference_t<@\exposid{maybe-const}@<OtherConst, V>>
+  operator-(const @\exposid{sentinel}@& y, const @\exposid{iterator}@<OtherConst>& x);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to: \tcode{return y.\exposid{end_} - x.\exposid{current_}.back();}
+\end{itemdescr}
+
+\rSec2[range.adjacent.transform]{Adjacent transform view}
+
+\rSec3[range.adjacent.transform.overview]{Overview}
+
+\pnum
+\tcode{adjacent_transform_view} takes an invocable object and
+a \libconcept{view} and produces a \libconcept{view}
+whose $M^\textrm{th}$ element is the result of applying the invocable object
+to the $M^\textrm{th}$ through $(M + N - 1)^\textrm{th}$ elements
+of the original view.
+If the original view has fewer than $N$ elements, the resulting view is empty.
+
+\pnum
+The name \tcode{views::adjacent_transform<N>} denotes
+a range adaptor object\iref{range.adaptor.object}.
+Given subexpressions \tcode{E} and \tcode{F} and
+a constant expression \tcode{N}:
```

Comment:
```suggestion
Given subexpressions \tcode{E} and \tcode{F}, and
a constant expression \tcode{N}:
```

Reply:
The status quo seems fine, and consistent with a previous use of "given...".